pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/domain.go

```diff
@@ -55,10 +55,11 @@ func GenerateDomainName(name string, obj metav1.ObjectMeta, ingressConfig *v1bet
 		return "", fmt.Errorf("error rendering the domain template: %w", err)
 	}
 
-	urlErrs := validation.IsFullyQualifiedDomainName(field.NewPath("url"), buf.String())
-	if urlErrs != nil {
-		return "", fmt.Errorf("invalid domain name %q: %w", buf.String(), urlErrs.ToAggregate())
+	if !ingressConfig.DisableIngressCreation {
```

Comment:
As far as I see this is going to skip the validation but not the generation so the method is going to return an invalid value in this case.

What about move this check where `GenerateDomainName` function is invoked to skip the invocation (while if the function is invoked the error is returned)?
My main concern is that this function `GenerateDomainName` in the case of `DisableIngressCreation=true` can return an invalid value with the assumption that it will not be used because of this configuration